quinn-proto/src/transport_error.rs

```diff
@@ -117,10 +116,7 @@ macro_rules! errors {
             fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
                 match self.0 {
                     $($val => f.write_str(stringify!($name)),)*
-                    x if x >= 0x100 && x < 0x200 => match AlertDescription::read_bytes(&[self.0 as u8]) {
-                        Some(desc) => write!(f, "Code::crypto({:?})", desc),
-                        None => write!(f, "Code::crypto({:02x})", self.0 as u8),
-                    },
+                    x if x >= 0x100 && x < 0x200 => write!(f, "Code::crypto({:02x})", self.0 as u8),
```

Comment:
I think this will make error messages such as certificate validation failures much less clear. Perhaps the crypto impl should provide a helper function to format its errors with?

Reply:
I'm not convinced this is an important issue. As far as I can see, the main issue left where there is no good human-readable error is for `ConnectionClose` frames received from the peer; which already can contain a human-readable description, so adding another one feels silly.

Second, if we were to do this I think the solution would get somewhat ugly, since these error codes get created in contexts where an implementation of the generic type is not easily available and it's non-trivial to pass one in for this `Debug` implementation.

So I feel that this is a reasonable trade-off for pluggable crypto.

One reasonable solution could be to use conditional compilation to improve the `Debug` representation only if `rustls` is enabled. We can always add the additional complexity for a fully generic solution once an actual alternative implementation comes along.

What do you think?

Reply:
> I'm not convinced this is an important issue. As far as I can see, the main issue left where there is no good human-readable error is for ConnectionClose frames received from the peer

Ah, I forgot we're already putting reasonable strings in there ourselves, good call. Of course callers interested in providing good user feedback will in any case want to match on the exact alert themselves, but this should be plenty to make development a pleasant experience.

> Second, if we were to do this I think the solution would get somewhat ugly, since these error codes get created in contexts where an implementation of the generic type is not easily available and it's non-trivial to pass one in for this Debug implementation.

We could use static dispatch to an associated function; no instance should be necessary. Of course, propagating a type parameter through to here would be a bit of a pain. Given the reason strings, I think the current code will do fine and we can revisit in the future if necessary.